subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/JavaTestProject.groovy

```diff
@@ -51,31 +53,63 @@ class JavaTestProject implements IncrementalTestProject {
 
     @Override
     void configure(CrossVersionPerformanceTestRunner runner) {
-        runner.gradleOpts.addAll(["-Xms${generator.daemonMemory}".toString(), "-Xmx${generator.daemonMemory}".toString()])
+        runner.gradleOpts.addAll(getMemoryOptions())
+    }
+
+    @Override
+    void configure(GradleBuildExperimentSpec.GradleBuilder builder) {
+        builder.invocation {
+            gradleOpts(memoryOptions)
+        }
+    }
```

Comment:
Not in this PR, but wouldn't it make sense to unify these APIs?

Reply:
Maybe. When we have a DSL for specifying the scenario, we should use the same DSL for regression and build comparisons.